src/index.ts

```diff
@@ -21,6 +22,89 @@ function writeLoaders(args: CLIArgs) {
     fs.writeFileSync(args.output, output);
 }
 
+/**
+ * If requiredList only contains responseKey
+ */
+function validRequiredList(requiredList: Array<string>, responseKey: string) {
+    if (requiredList.length > 1) return false;
+    if (requiredList.length == 1 && requiredList[0] !== responseKey) {
+        return false;
+    }
+    return true;
+}
+
+/**
+ * Find all values of the intput key in an object recursively
+ */
+function findVal(object: any, key: string, array: Array<string>) {
+    Object.keys(object).some(function (k) {
+        if (k === key) {
+            array.push(...object[k]);
+        }
+        if (object[k] && typeof object[k] === 'object') {
+            findVal(object[k], key, array);
+        } else {
+            return;
+        }
+    });
+    return array;
+}
+
+/**
+ * Throw erros when resource uses propertyBatchKey but doesn't match requirements
+ */
+function verifyBatchPropertyResource(args: CLIArgs) {
```

Comment:
As discussed, since this is quite Yelpy and relies on checking swagger stuff specifically (which this tool is currently opinionated about), I'd suggest keeping this logic but moving it to a seperate script that runs in our build, outside of this tool.

That way, we can remove the swaggerPath / httpMethod keys etc since we'll know how to map to it.

Reply:
Hmm, you mean move the script to ggs? But the `propertyBatchKey` needs to be in dataloader-codegen, if anyone else uses `propertyBatchKey` they won't have correct flow type.

Reply:
nvm, saw your other command!